compiler/executable/match_/planner/vertex/mod.rs

```diff
@@ -459,7 +459,13 @@ impl<'a> DisjunctionPlanner<'a> {
     ) -> Self {
         let shared_variables =
             builder.branches().iter().flat_map(|pb| pb.shared_variables()).map(|v| variable_index[v]).collect();
```

Comment:
This should maybe also have a `dedup()` ?

Reply:
I'll put a type hint that it's a `HashSet<_>` so it breaks if the type changes.